plugins/utils/tmux-navigator.nix

```diff
@@ -6,37 +6,126 @@
   ...
 }:
 with lib; let
-  cfg = config.plugins.tmux-navigator;
-in {
-  options.plugins.tmux-navigator = {
-    enable = mkEnableOption "Tmux-Navigator (see https://github.com/christoomey/vim-tmux-navigator for tmux installation instruction)";
-
-    package = helpers.mkPackageOption "tmux-navigator" pkgs.vimPlugins.tmux-navigator;
-
-    tmuxNavigatorSaveOnSwitch = helpers.mkNullOrOption (lib.types.enum [1 2]) ''
-      null: don't save on switch (default value)
-      1: update (write the current buffer, but only if changed)
-      2: wall (write all buffers)
-    '';
-
-    tmuxNavigatorDisableWhenZoomed = helpers.mkNullOrOption (lib.types.enum [1]) ''
-      null: unzoom when moving from Vim to another pane (default value)
-      1: If the tmux window is zoomed, keep it zoomed when moving from Vim to another pane
-    '';
-
-    tmuxNavigatorNoWrap = helpers.mkNullOrOption (lib.types.enum [1]) ''
-      null: move past the edge of the screen, tmux/vim will wrap around to the opposite side (default value)
-      1: disable wrap
-    '';
-  };
-
-  config = mkIf cfg.enable {
-    extraPlugins = [cfg.package];
-
-    globals = {
-      tmux_navigator_save_on_switch = cfg.tmuxNavigatorSaveOnSwitch;
-      tmux_navigator_disable_when_zoomed = cfg.tmuxNavigatorDisableWhenZoomed;
-      tmux_navigator_no_wrap = cfg.tmuxNavigatorNoWrap;
+  # Added 2024-03-19
+  deprecations = let
+    pluginPath = ["plugins" "tmux-navigator"];
+    option = s: pluginPath ++ [s];
+    setting = s: pluginPath ++ ["settings" s];
+    settingStr = s: concatStringsSep "." (setting s);
+  in [
+    (
+      mkRenamedOptionModule
+      (option "tmuxNavigatorSaveOnSwitch")
+      (setting "save_on_switch")
+    )
+    (
+      mkRemovedOptionModule
+      (option "tmuxNavigatorDisableWhenZoomed")
+      "Use `${settingStr "disable_when_zoomed"}` option."
+    )
+    (
+      mkRemovedOptionModule
+      (option "tmuxNavigatorNoWrap")
+      "Use `${settingStr "no_wrap"}` option."
+    )
+  ];
+in
+  helpers.vim-plugin.mkVimPlugin config {
+    # FIXME document tmux-side installation instructions
+    name = "tmux-navigator";
+    originalName = "vim-tmux-navigator";
+    defaultPackage = pkgs.vimPlugins.tmux-navigator;
+    globalPrefix = "tmux_navigator_";
+
+    maintainers = [maintainers.GaetanLepage];
+
+    imports = deprecations;
+
+    settingsOptions = {
+      # FIXME can this be enum of ["buffer" "all"]?
+      save_on_switch = helpers.mkNullOrOption (types.enum [1 2]) ''
+        You can configure the plugin to write the current buffer, or all buffers, when navigating from Vim to tmux.
```

Comment:
`vim`, `Vim`, `tmux` capitalization is a little inconsistent.

Reply:
I would go for "vim" here, but this is not too important.